lib/internal/fs/promises.js

```diff
@@ -917,6 +920,25 @@ async function readFile(path, options) {
   return handleFdClose(readFileHandle(fd, options), fd.close);
 }
 
+function _watch(filename, options = kEmptyObject) {
+  validateObject(options, 'options');
+
+  if (options.recursive != null) {
+    validateBoolean(options.recursive, 'options.recursive');
+
+    // TODO(anonrig): Remove this when/if libuv supports it.
+    // libuv does not support recursive file watch on Linux due to
+    // the limitations of inotify.
+    if (options.recursive && !isOSX && !isWindows) {
+      const watcher = new nonNativeWatcher.FSWatcher(options);
+      watcher[kFSWatchStart](filename);
+      return watcher[SymbolAsyncIterator]();
```

Comment:
If we're making an AsyncGenerator
```suggestion
      yield* watcher;
      return;
```